File: solutions/136/main.py
from typing import List 

class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        result = 0
        for n in nums:
            result ^= n
        return result

if __name__ == "__main__":
    nums = [2,2,1]
    s = Solution()
    assert s.singleNumber(nums) == 1

File: solutions/136/main.cpp
#include <iostream>
#include <vector>
#include <gtest/gtest.h> 

using namespace std; 

class Solution {
public:
    // time complexity O(n)
    // space complexity O(1)
    int singleNumber(vector<int>& nums) {
        int x = 0; 
        for (const auto i : nums)
        {
            x ^= i; 
        }
        return x; 
    }
};

TEST(Test136, SimpleTest)
{
    vector<int> nums = {2,2,1};
    Solution s; 
    EXPECT_EQ(s.singleNumber(nums), 1) << "wrong answer";
}

int main()
{
    testing::InitGoogleTest();
    return RUN_ALL_TESTS();
}
